File: src/linkerz_job_queue/src/main/java/org/linkerz/job/queue/handler/Handler.java
/*
 * Copyright (C) 2012 - 2013 LinkerZ
 */

package org.linkerz.job.queue.handler;

import org.linkerz.job.queue.job.Job;
import org.linkerz.job.queue.session.Session;

/**
 * The Class Handler.
 *
 * @author Nguyen Duc Dung
 * @since 7/4/12, 2:01 PM
 */
public interface Handler<J extends Job, S extends Session> {

    /**
     * Check whether this handler is for this class or not.
     *
     * @param clazz
     * @return
     */
    boolean isFor(Class<J> clazz);

    /**
     * Handle the job.
     *
     * @param job
     * @param session
     */
    void handle(J job, S session) throws Exception;
}

File: src/linkerz_job_queue/src/main/scala/org/linkerz/job/queue/core/Handler.scala
/*
 * Copyright (C) 2012 - 2013 LinkerZ
 */

package org.linkerz.job.queue.core

/**
 * The Class Worker.
 *
 * @author Nguyen Duc Dung
 * @since 7/7/12, 10:33 PM
 *
 */

/**
 * Handle of the job.
 * @tparam J job
 */
trait Handler[J <: Job] {

  /**
   * Check is this worker is for the job or not.
   * @param job
   * @return false if it is not.
   */
  def accept(job: Job): Boolean

  /**
   * Handel for the job in session.
   * @param job
   * @param session
   */
  def handle(job: Job, session: Session) {
    doHandle(job.asInstanceOf[J], session)
  }

  /**
   * Handel the job
   * @param job
   * @param session
   */
  protected def doHandle(job: J, session: Session)

}